OOPConsoleApp/Program.cs

```diff
@@ -0,0 +1,313 @@
+
+                Console.WriteLine($"{person.GetSurname()} " +
+                    $"{person.GetName()}, " +
+                    $"{person.GetAge()}");
+            }
+
+            void Exercise2()
+            {
+                Console.WriteLine("Exercise 2.");
+
+                var student = new Student("Koval", "Irina", 19);
+                student.SetStudentAbility("musical ear");
+                student.SetStudentAssesment(5);
+
+                Console.WriteLine($"Student - {student.PersonSurname}" +
+                    $" {student.PersonName}," +
+                    $" {student.PersonAge} years old.");
+                Console.WriteLine($"Ability: {student.StudentAbility}. Assesment: {student.StudentAssesment}.");
+
+                var teacher = new Teacher("Ivanov", "Ivan", 41);
+                teacher.SetTeacherTemperament("sanguine");
+                teacher.SetTeacherSubject("Mathematicks");
+
+                Console.WriteLine($"Teacher - {teacher.PersonSurname} " +
+                    $"{teacher.PersonName}, " +
+                    $"{teacher.PersonAge} years old. ");
+                Console.WriteLine($"Temperament: {teacher.TeacherTemperament}. Subject: {teacher.TeacherSubject}.");
+            }
+            void Exercise3()
+            {
+                Console.WriteLine("Exercise 3.");
+
+                var personSecond = new Person("Datchin", "Valery", 43);
+
+                Console.WriteLine($"{personSecond.GetSurname()} {personSecond.GetName()}," +
+                $" {personSecond.GetAge()} y.o.");
+            }
+            void Exercise4()
+            {
+                Console.WriteLine("Exercise 4.");
+
+                var myPerson = new Person("Fleming", "Lise", 33);
+                myPerson.Print();
+
+                var myStudent = new Student("Koval", "Irina", 19);
+                myStudent.Print();
+
+                var myTeacher = new Teacher("Ivanov", "Ivan", 41);
+                myTeacher.Print();
+            }
+
+            void Exercise5()
+            {
+                Console.WriteLine("Exercise 5.");
+
+                var persons = new List<Person>
+                {
+                    { new Student("Pupkin", "Vasiliy", 20)},
+                    { new Student("Vasilyeva", "Iren", 18)},
+                    { new Teacher("Kirkorov", "Petya", 46)},
+                };
+
+                var classroom = new Classroom();
+                foreach (var person in persons)
+                {
+                    classroom.AddPerson(person);
+                }
+
+                classroom.Print();
+
+                foreach (var person in persons)
+                {
+                    classroom.RemovePerson(person);
+                }
+
+                classroom.ClearPersons();
+            }
+        }
+    }
+}
+
+namespace OOPConsoleApp
+{
+    public class Classroom : IPrintable
+    {
+        private readonly Dictionary<int, Person> _persons = new();
```

Comment:
you don't need to use a dictionary everywhere, especially if you don't need to store the key, because This is an index, it is already stored in an array and a list.